.github/workflows/testing.yml

```diff
@@ -0,0 +1,43 @@
+name: Testing
+
+on: [push]
```

Comment:
suggestion:

```yaml

on: [push, pull_request]

concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true

permissions:
  contents: read # to fetch code (actions/checkout)
```

* `on: [push, pull_request]` - to run on PR as well
* the concurrency bit is to allow cancelling a running pipeline if you push new changes
* the permissions one is to tighten up what the CI can do

Reply:
This is now updated